webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioBuilderDialog/utils.ts

```diff
@@ -2,35 +2,177 @@ import { AxiosResponse } from "axios";
 import { StudyMetadata } from "../../../../../../../../common/types";
 import client from "../../../../../../../../services/api/client";
 
-export const TABS_DATA: Array<[string, string]> = [
-  ["load", "l"],
-  ["thermal", "t"],
-  ["hydro", "h"],
-  ["wind", "w"],
-  ["solar", "s"],
-  ["ntc", "ntc"],
-  ["hydroLevels", "hl"],
-];
+////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////
 
-export const ACTIVE_SCENARIO_PATH =
-  "settings/generaldata/general/active-rules-scenario";
+export const SCENARIOS = [
+  "load",
+  "thermal",
+  "hydro",
+  "wind",
+  "solar",
+  "ntc",
+  "renewable",
+  "bindingConstraints",
+] as const;
 
-export type ScenarioBuilderConfig = Record<string, unknown>;
+export type ScenarioType = (typeof SCENARIOS)[number];
 
-function makeRequestURL(studyId: StudyMetadata["id"]): string {
-  return `v1/studies/${studyId}/config/scenariobuilder`;
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+// Represents values that can be either a number or an uninitialized string (rand).
+export type YearlyValues = number | "";
+export type ElementConfig = Record<string, YearlyValues>;
+// General configuration format for scenarios using areas as element IDs.
+export type GenericScenarioConfig = Record<string, ElementConfig>;
+
+export type ClusterConfig = Record<string, YearlyValues>;
```

Comment:
What the purpose to have duplicate types?

Reply:
I could use `ElementConfig` for `ClusterConfig` but its less descriptive what the type is for

Reply:
```typescript
export type ElementConfig = Record<string, YearlyValues>;
export type ClusterConfig = Record<string, YearlyValues>;
```
=> 
```typescript
export type Config = Record<string | number, YearlyValues>;
```
---
```typescript
export type GenericScenarioConfig = Record<string, ElementConfig>;
export type AreaClustersConfig = Record<string, ClusterConfig>;
```
=> 
```typescript
// Items can be areas or clusters
export type ConfigByItems = Record<string, YearlyValues>;
```

Reply:
No, the keys are always strings representing an index that maps to a `mc_year` value that is a number, or an empty string (default for empty configuration that results in "rand") 
example:
```json
{
	"load": {
		"z1": {
			"0": 15,
			"1": "",
			"2": "",
			"3": "",
			"4": "",
			"5": "",
			"6": "",
			"7": "",
			"8": "",
			"9": "",
			"10": "",
			"11": "",
			"12": "",
			"13": "",
			"14": ""
		}
	}
}
```
=> Generic config

```json
{
	"thermal": {
		"z1": {
			"cluster1": {
				"0": 10,
				"1": "",
				"2": "",
				"3": "",
				"4": "",
				"5": 48,
				"6": "",
				"7": "",
				"8": "",
				"9": "",
				"10": "",
				"11": "",
				"12": "",
				"13": "",
				"14": ""
			},
			"cluster2": {
				"0": 12,
				"1": "",
				"2": "",
				"3": "",
				"4": 554,
				"5": "",
				"6": "",
				"7": "",
				"8": "",
				"9": "",
				"10": "",
				"11": "",
				"12": "",
				"13": "",
				"14": ""
			}
		},

	}
}
```

=> Nested clusters configuration

Reply:
Surely it can be simplified, but I want to keep this distinction clear